README.md

```diff
@@ -13,46 +13,133 @@ Features
+cd wagtail/
+# 4. Run the setup script. This will set up all required dependencies for you.
+./setup.sh
+```
+
+> Note: On platforms that can't run shell scripts, run the commands from [`setup.sh`](setup.sh) manually instead.
+
+Here is the resulting folder structure:
+
+```sh
+.
+├── libs          # Libs to develop wagtail against.
+├── vagrant       # Vagrant-related files.
+├── wagtail       # Wagtail repository / codebase.
+└── wagtaildemo   # wagtaildemo project used for development.
+```
+
+Once setup is over,
+
+```sh
+# 5. ssh into your new Vagrant virtual machine.
+vagrant ssh
+# 6. Start up the wagtaildemo development server.
+./manage.py runserver 0.0.0.0:8000
+# Success!
+```
+
+- Visit your site at http://localhost:8000
+- The admin interface is at http://localhost:8000/admin/ - log in with `admin` / `changeme`.
 
 What you can do
 ---------------
 
-Start up wagtaildemo:
-
-    ./manage.py runserver 0.0.0.0:8000
+> Note: all of those commands are meant to be used **inside the Vagrant virtual machine**. To get there, go to your local Wagtail (`cd ~/Development/wagtail`) set up and `vagrant up` then `vagrant ssh`.
 
-and visit http://localhost:8000. The admin interface is at http://localhost:8000/admin/ - log in with admin / changeme
+Start the wagtaildemo server:
 
-Run tests:
+```sh
+./manage.py runserver 0.0.0.0:8000
+# Then visit http://localhost:8000 in your browser.
+```
 
-    cd ../wagtail
-    ./runtests.py
+Run the tests:
 
-Build front-end assets:
+```sh
+cd /home/vagrant/wagtail
+# Python tests.
+./runtests.py
+# Node tests.
+npm run test
+```
 
-    cd ../wagtail
-    npm run build
+Run the linting:
 
-(or `npm run watch` to watch the source files for changes)
+```sh
+cd /home/vagrant/wagtail
+# Python linting.
+# TODO
+# JavaScript linting.
+npm run lint
+```
 
-Build the docs:
-
-    cd ../wagtail/docs
-    make html
-
-Switch to Python 2.7:
-
-    source ~/.virtualenvs/wagtailpy2/bin/activate
-
-And back to Python 3:
+Build front-end assets:
 
-    source ~/.virtualenvs/wagtaildemo/bin/activate
+```sh
+cd /home/vagrant/wagtail
+npm run build
+# Note: can also be done from outside the virtual machine, provided you have the right version of node installed.
+```
+
+Start front-end development tools and file watching:
+
+```sh
+cd /home/vagrant/wagtail
+npm run start
+# Note: can also be done from outside the virtual machine, provided you have the right version of node installed.
+```
+
+Build the documentation:
+
+```sh
+cd /home/vagrant/wagtail/docs
+make html
+```
+
+Switch between Python versions (Wagtail supports multiple ones):
+
+```sh
+# To switch to python 2.7,
+source ~/.virtualenvs/wagtailpy2/bin/activate
+# And to switch back to Python 3,
+source ~/.virtualenvs/wagtaildemo/bin/activate
+```
+
+Getting ready to contribute
+---------------------------
+
+Here are other actions you will likely need to do to make your first contribution to the project.
+
+Set up git remotes to Wagtail forks:
```

Comment:
We should make it clear that this needs to be done outside of the VM (or else you need to fiddle with SSH keys and things).